app/domain/authentication/authn_oidc/update_input_with_username_from_id_token.rb

```diff
@@ -53,9 +52,17 @@ def verify_and_decode_token
         )
       end
 
-      # The credentials are in a URL encoded form data in the request body
+      # The credentials are in a URL encoded form data in the request body or in the request header
       def decoded_credentials
```

Comment:
Hey @aloncarmel111 , this is some tricky logic to follow, what do you think about something like this instead?

```ruby
      # The credentials are in a URL encoded form data in the request body or in the request header
      def decoded_credentials
        @decoded_credentials ||= begin
          return token_from_header if token_from_header

          return token_from_body if token_from_body

          # If the token is not in the header or body, raise a missing param exception
          raise Errors::Authentication::RequestBody::MissingRequestParam, 'id_token'
        end
      end

      def token_from_header
        @token_from_header ||= begin
          return nil unless request&.headers&.key?("HTTP_AUTHORIZATION")

          # Extract the token from the authorization header
          id_token = request.headers['HTTP_AUTHORIZATION'].split(' ', -1)

          # Verify the token is given as a bearer token
          return nil unless id_token[0] == "Bearer" && id_token.length == 2

          # Return the token formatted as it would be if extracted from the body
          # as `application/x-www-form-urlencoded` data.
          {
            # URL decode the ID token from the header
            "id_token" => URI.decode_www_form_component(id_token[1])
          }
        end
      end

      def token_from_body
        @token_from_body ||= begin
          # Parse the request body
          credential_token = Hash[URI.decode_www_form(credentials)]

          # Return the parsed body if it include the token
          credential_token unless credential_token.fetch('id_token', '').empty?
        end
      end
```